libfreerdp/codec/dsp.c

```diff
@@ -768,9 +772,15 @@ static BOOL freerdp_dsp_encode_ima_adpcm(FREERDP_DSP_CONTEXT* context, const BYT
 			*dst++ = encoded;
 			size -= 4;
 		}
+
+		if (dst - start == context->adpcm.ima.packet_size)
+		{
+			Stream_Write(out, start, context->adpcm.ima.packet_size);
+			dst = Stream_Buffer(context->buffer);
```

Comment:
sure you want to ignore all data after `packet_size`?

Reply:
No, I don't ignore the data. The encoded data is still in context->buffer and will be recovered in the next call to `freerdp_dsp_encode_ima_adpcm`

Reply:
Ok, I see that now. The way you did it confused me a bit. Maybe better use `Stream_GetPosition` and just do a `Stream_SetPosition(buffer, 0)` on write?

Reply:
Ok, see why you did it this way now, ok, we can leave it as is.